src/App.tsx

```diff
@@ -1,31 +1,41 @@
 import SwatchWithForm from './SwatchWithForm';
 import { StitchPattern, Color } from './types'
+import { useState, useEffect } from "react";
+import { useSearchParams } from 'react-router-dom';
+import { URLSearchParamsFromSwatchParams, sanitizeSearchParamInputs } from './searchHelpers';
 
 const red = "#ff001d" as Color;
 const cream = "#fcf7eb" as Color;
 const ltblue = "#8dd0f2" as Color;
 const navy = "#0e0e66" as Color;
 
 function App() {
-  const config = {
-    colorConfig: [
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [swatchParams, setSwatchParams] = useState({
+    colorConfig: sanitizeSearchParamInputs.colorConfig(searchParams) || [
       {color: navy, length: 3},
       {color: red, length: 3},
       {color: navy, length: 3},
       {color: ltblue, length: 2},
       {color: cream, length: 5},
       {color: ltblue, length: 2},
     ],
-    crowLength: 18,
-    crows: 40,
-    colorShift: 0,
-    staggerLengths: false,
-    stitchPattern: StitchPattern.moss,
+    crowLength: sanitizeSearchParamInputs.crowLength(searchParams) || 18, //Note: explicitly ok not saving zero from search params here
+    crows: sanitizeSearchParamInputs.crows(searchParams) || 40, //Note: explicitly ok not pulling zero from search params here
+    colorShift: sanitizeSearchParamInputs.colorShift(searchParams) || 0,
+    staggerLengths: sanitizeSearchParamInputs.staggerLengths(searchParams),
+    stitchPattern: sanitizeSearchParamInputs.stitchPattern(searchParams) || StitchPattern.moss,
     showRowNumbers: false
```

Comment:
I decided this is strictly a view concern and does not need to come from the URL. I might change this value so that it doesn't end up above SwatchWithForm.

Reply:
Ok did that change in another PR